### DISCUSSION TITLE
mDNS does not use name from config -- resolved

### DISCUSSION THREAD

**Discussion by trip5** [EdenRidgway/XY-Wifi-Clock#10] [Ideas]
In the older version it still used xyclock.local and with the update it uses xy-clock.local consistently. That seems to indicate that it is not using the variables pulled from the config file during the setup phase. There's nothing to do about that, is there?

In some way, that explains to me why Tasmota never used mDNS but ESPHome devices can...  The name needs to be compiled in doesn't it?

**Comment by EdenRidgway** [comment]
The line below should mean that it is uses the default device name or the one set by the user:
```c
if (MDNS.begin(config.getDeviceName())) {
```

The current default should be xyclock but it used to be xy-clock (so I'm not sure why it would now be xy-clock):
```c
if (!config.getDeviceName()) {
    config.setDeviceName("XYClock");
}
```

**Reply by trip5** [reply]
I did change that part of the code to add the hyphen in multiple places where there wasn't one before... it seemed to be a bit inconsistent across various files.  With the hyphen just seemed more common (and it's part of the Github name) so I put it in where it was missing... with the update it responds to pings at XY-Clock.local just fine... Do you think it's better without?

But yeah, I get what that line means, I just can't understand why the config isn't being used here. I think it may be getting the mDNS name from this section:

            String getDeviceName() {
            if (deviceName == NULL || deviceName == "") {
                deviceName = "XY-Clock";
            }

I just flashed one with some test names in various places.  But if I remember, it takes some time for the mDNS cache to expire so I might have to wait a bit to confirm my suspicion.

**Reply by EdenRidgway** [reply]
Yeah, making it consistently xy-clock makes sense. I've also been finding mDNS unreliable so I've switched to using the IP address instead.

**Comment by trip5** [comment]
I found the OTA code had a recommended line that sets the name to be used in Arduino and I figured why not use the config's Device Name but it wouldn't compile unless converted to a c string... what a mess of brackets!

ArduinoOTA.setHostname((config.getDeviceName()).c_str())

And that's the line that can set the mDNS name for real!  Weird.  But works.  I added restart code to the Website for changing the Device Name as well.  And there's almost no delay for a ping to the mDNS name after a reboot.  It does respond to the old name for a bit.

**Comment by trip5** [comment]
Just thought I'd put here that my mDNS on Windows 11 was acting weird, even for my ESPHome devices so I installed Apple Bonjour from here:  https://support.apple.com/kb/dl999?locale=en_US -- supposedly this is the easiest way to resolve a known issue with Windows...
